### PR TITLE
Add `addEventListener()` typings to `MediaRecorder`

### PR DESCRIPTION
**Pull request by AppLover69**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://developer.mozilla.org/en-US/docs/Web/API/MediaRecorder>

### CODE REVIEW

File: types/dom-mediacapture-record/index.d.ts [hunk 1]
```diff
@@ -48,6 +49,13 @@ declare class MediaRecorder extends EventTarget {
 
     constructor(stream: MediaStream, options?: MediaRecorderOptions);
 
+    addEventListener(type: 'dataavailable', listener: (this: MediaRecorder, event: BlobEvent) => void, useCapture?: boolean): void;
```
Comment (eliasmeire) [inline types/dom-mediacapture-record/index.d.ts]: I don't think this is needed. MediaRecorder already extends EventTarget which has the correct type for `addEventListener`.

**Review by elsmr — CHANGES_REQUESTED** [review]

File: types/dom-mediacapture-record/index.d.ts (hunk 1 above)
Comment (AppLover69) [reply types/dom-mediacapture-record/index.d.ts]: Right, but it lacks the typings for `'dataavailable'` and `'error'` events

Comment (eliasmeire) [reply types/dom-mediacapture-record/index.d.ts]: Ah I see, good improvement! maybe you could mimic the way TypeScript does it: https://github.com/microsoft/TypeScript/blob/master/lib/lib.dom.d.ts#L2381-L2400. They use an EventMap to enforce both type and event.

Comment (AppLover69) [reply types/dom-mediacapture-record/index.d.ts]: That's what I did locally in my own projects, I'll gladly do so 😄

**Review by elsmr — APPROVED** [review]

**Pull request merged**